### PR TITLE
request body will not be set to null by default.

### PR DESCRIPTION
**Pull request by amarzavery**

### CODE REVIEW

**Comment by amarzavery** [comment]
@thecodejedi - This should fix https://github.com/Azure/ms-rest-js/issues/15. 
I believe this PR https://github.com/Azure/ms-rest-js/pull/16 would not be required.

**Comment by olydis** [comment]
# 🤖 AutoRest automatic publish job 🤖
~~~ Haskell
> will publish once PR gets merged
~~~

File: src/vanilla/Templates/MethodTemplate.cshtml [hunk 1]
```diff
@@ -142,12 +142,6 @@ async @(Model.Name.ToCamelCase())WithHttpOperationResponse(@(Model.MethodParamet
   httpRequest.formData = formData;
   </text>      
   }
-  else
-  {
-  <text>
-  httpRequest.body = null;
-  </text>
-  }
```
Comment (amarzavery) [inline src/vanilla/Templates/MethodTemplate.cshtml]: @thecodejedi - This change in the generator should fix your problem. Can you also verify by removing `httpRequest.body = null;` in the generated client for `GET` and `HEAD` methods and test whether it works for you. Make sure that you have reverted your changes in the local copy of `ms-rest-js` while testing this change.

**Comment by olydis** [comment]
# 🤖 AutoRest automatic feature coverage report 🤖
*feature set version 2.3.22*

## ❌️ General: 96%

<details><summary>20 out of 528 features are not covered by tests</summary><p>

[`HttpRedirect301Put`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect301Put), [`HttpRedirect302Patch`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect302Patch), [`UrlQueriesArrayMultiEmpty`](https://github.com/Azure/autorest.testserver/search?q=UrlQueriesArrayMultiEmpty), [`UrlQueriesArrayMultiNull`](https://github.com/Azure/autorest.testserver/search?q=UrlQueriesArrayMultiNull), [`UrlQueriesArrayMultiValid`](https://github.com/Azure/autorest.testserver/search?q=UrlQueriesArrayMultiValid), [`additionalPropertiesInProperties`](https://github.com/Azure/autorest.testserver/search?q=additionalPropertiesInProperties), [`additionalPropertiesInPropertiesWithAPTypeString`](https://github.com/Azure/autorest.testserver/search?q=additionalPropertiesInPropertiesWithAPTypeString), [`additionalPropertiesTrue`](https://github.com/Azure/autorest.testserver/search?q=additionalPropertiesTrue), [`additionalPropertiesTypeObject`](https://github.com/Azure/autorest.testserver/search?q=additionalPropertiesTypeObject), [`additionalPropertiesTypeString`](https://github.com/Azure/autorest.testserver/search?q=additionalPropertiesTypeString), [`allowedValueEnum`](https://github.com/Azure/autorest.testserver/search?q=allowedValueEnum), [`expectedEnum`](https://github.com/Azure/autorest.testserver/search?q=expectedEnum), [`getEnumReferenced`](https://github.com/Azure/autorest.testserver/search?q=getEnumReferenced), [`getEnumReferencedConstant`](https://github.com/Azure/autorest.testserver/search?q=getEnumReferencedConstant), [`putComplexPolymorphismComplicated`](https://github.com/Azure/autorest.testserver/search?q=putComplexPolymorphismComplicated), [`putComplexPolymorphismNoDiscriminator`](https://github.com/Azure/autorest.testserver/search?q=putComplexPolymorphismNoDiscriminator), [`putEnumReferenced`](https://github.com/Azure/autorest.testserver/search?q=putEnumReferenced), [`putEnumReferencedConstant`](https://github.com/Azure/autorest.testserver/search?q=putEnumReferencedConstant), [`roundTripEnum`](https://github.com/Azure/autorest.testserver/search?q=roundTripEnum), [`unexpectedEnum`](https://github.com/Azure/autorest.testserver/search?q=unexpectedEnum)</p></details>

## ❌️ Azure: 99%

<details><summary>1 out of 118 features are not covered by tests</summary><p>

[`LRONonRetryPut201Creating400InvalidJson`](https://github.com/Azure/autorest.testserver/search?q=LRONonRetryPut201Creating400InvalidJson)</p></details>

File: src/vanilla/Templates/MethodTemplate.cshtml (hunk 1 above)
Comment (thecodejedi) [reply src/vanilla/Templates/MethodTemplate.cshtml]: @amarzavery Manually removed it after generating the code and restored original ms-rest-js, works like a charm on Firefox 45 now!

I already closed Azure/ms-rest-js#16 as it does fix Azure/ms-rest-js#15.

In which version of autorest will this change be available?

Comment (amarzavery) [reply src/vanilla/Templates/MethodTemplate.cshtml]: `2.0.87` is the latest version of `@microsoft.azure/autorest.typescript`

Comment (amarzavery) [reply src/vanilla/Templates/MethodTemplate.cshtml]: try `autorest --reset` and then execute the command to generate the client. It should bring the latest version while generating the client. Post generation try `autorest --info` to verify that you have the latest version.

Comment (amarzavery) [reply src/vanilla/Templates/MethodTemplate.cshtml]: at anytime if you want to use a preview or a specific version you can say `--use=@microsoft.azure/autorest.typescript@2.0.77`. If you want to use a local build while you are developing the generator you can say `--use=<absolute-path-to/autorest.typescript>`

**Pull request merged**